src/components/Swap/Settings/index.tsx

```diff
@@ -41,13 +43,23 @@ export default function Settings() {
 
   useOnEscapeHandler(() => setOpen(false))
 
+  const isMobile = useIsMobileWidth()
   return (
     <div ref={setWrapper}>
-      <PopoverBoundaryProvider value={wrapper}>
-        <Popover showArrow={false} offset={10} show={open} placement="top-end" content={<SettingsPopover />}>
+      {isMobile ? (
+        <>
           <SettingsButton onClick={() => setOpen(!open)} icon={SettingsIcon} />
-        </Popover>
-      </PopoverBoundaryProvider>
+          <BottomSheetModal title="Settings" onClose={() => setOpen(false)} open={open}>
+            <SettingsPopover />
```

Comment:
nit: Should this now be named something more generic than `SettingsPopover`?

Reply:
this is specific to settings tho!

Reply:
i just understood what you meant lol. renamed in a diff PR!